streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopicConfig.java

```diff
@@ -67,33 +57,24 @@ private boolean isCompactDelete() {
      * @param additionalRetentionMs - added to retention to allow for clock drift etc
      * @return Properties to be used when creating the topic
      */
-    public Properties toProperties(final long additionalRetentionMs) {
-        final Properties result = new Properties();
-        for (Map.Entry<String, String> configEntry : logConfig.entrySet()) {
-            result.put(configEntry.getKey(), configEntry.getValue());
-        }
-        if (retentionMs != null && isCompactDelete()) {
-            result.put(InternalTopicManager.RETENTION_MS, String.valueOf(retentionMs + additionalRetentionMs));
-        }
-
-        if (!logConfig.containsKey(InternalTopicManager.CLEANUP_POLICY_PROP)) {
-            final StringBuilder builder = new StringBuilder();
-            for (CleanupPolicy cleanupPolicy : cleanupPolicies) {
-                builder.append(cleanupPolicy.name()).append(",");
-            }
-            builder.deleteCharAt(builder.length() - 1);
+    public Map<String, String> toProperties(final long additionalRetentionMs) {
```

Comment:
Nit: Return type and method name to does match. maybe rename to `addToTopicConfig` ?

Also update JavaDoc `return` tag

Reply:
I think the name / return tag still makes sense: it returns a map of properties from its `topic config`, `retention ms` and the passed-in `additional retention ms`.

Reply:
Sound like it uses `Properties` class (while it doesn't) -- but maybe I am overthinking this... Should be fine.